consensus/election/election_test.go

```diff
@@ -0,0 +1,121 @@
+package election
+
+import (
+	"github.com/nknorg/nkn/v2/common"
+	"github.com/nknorg/nkn/v2/config"
+	"os"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+)
+
+var c = &Config{
+	Duration:                    config.ConsensusDuration / 2,
+	MinVotingInterval:           200 * time.Millisecond,
+	MaxVotingInterval:           2 * time.Second,
+	ChangeVoteMinRelativeWeight: 0.5,
+	ConsensusMinRelativeWeight:  2.0 / 3.0,
+}
+
+func TestMain(m *testing.M) {
+	os.Exit(m.Run())
+}
+
+func TestSingleElection(t *testing.T) {
+	check := func(f string, got, want interface{}) {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("%s mismatch: got %v, g %v", f, got, want)
+		}
+	}
+
+	elc, err := NewElection(c)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = elc.SetInitialVote(common.MaxUint256)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = elc.PrefillNeighborVotes([]interface{}{"100", "200", "300"}, common.EmptyUint256)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	elc.Start()
+
+	elc.Stop()
+
+	result, _, _, err := elc.GetResult()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	check("result", result, common.EmptyUint256)
+}
+
+func TestMultiElection(t *testing.T) {
+	check := func(f string, got, want interface{}) {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("%s mismatch: got %v, g %v", f, got, want)
+		}
+	}
+
+	v1 := common.EmptyUint256
+	v1[0] = 1
+
+	elcList := make([]*Election, 20)
+
+	var err error
+	for i := 0; i < len(elcList); i++ {
+		elcList[i], err = NewElection(c)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = elcList[i].SetInitialVote(common.EmptyUint256)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var v common.Uint256
+		if i < 16 {
+			v = v1
+		} else {
+			v = common.MaxUint256
+		}
+		err = elcList[i].PrefillNeighborVotes([]interface{}{"100", "200", "300"}, v)
```

Comment:
This is not right. Each election should know the correct id and vote of all other nodes.